deployment_scripts/deploy_linux.sh

```diff
@@ -63,6 +63,10 @@ log_message "loaded configuration"
 
 # Setup monkey either in dir required or current dir
 monkey_home=${1:-$(pwd)}
+
+# Replace front ~ with the home directory
+monkey_home=${monkey_home/#'~'/$HOME}
```

Comment:
This is an incomplete solution. A path like "~/my-stuff" will evaluate to "$HOME/my-stuff", but a path like "~user/my-stuff" should evaluate to "/home/user/my-stuff". Using string substitution results in "$HOMEuser/my-stuff".

Good work in tracking down the source of the issue, though. IMHO, this is actually a non-issue. If the tilde is quoted, then the script behaves as expected. I'll leave some comments on the original issue.